src/chunk.c

```diff
@@ -970,34 +972,25 @@ chunk_create_metadata_after_lock(Hypertable *ht, Hypercube *cube, const char *sc
 	else
 		namestrcpy(&chunk->fd.table_name, table_name);
 
-	/* Insert chunk */
-	ts_chunk_insert_lock(chunk, RowExclusiveLock);
-
-	/* Insert any new dimension slices */
-	ts_dimension_slice_insert_multi(cube->slices, cube->num_slices);
-
-	/* Add metadata for dimensional and inheritable constraints */
-	chunk_add_constraints(chunk);
-	ts_chunk_constraints_insert_metadata(chunk->constraints);
-
-	/* If this is a remote chunk we assign data nodes */
 	if (chunk->relkind == RELKIND_FOREIGN_TABLE)
 		chunk->data_nodes = chunk_assign_data_nodes(chunk, ht);
 
 	return chunk;
 }
 
-static Oid
-chunk_create_table_after_lock(Chunk *chunk, Hypertable *ht)
+static void
+chunk_insert_into_metadata_after_lock(const Chunk *chunk)
 {
-	/* Create the actual table relation for the chunk */
-	const char *tablespace = ts_hypertable_select_tablespace_name(ht, chunk);
-
-	chunk->table_id = ts_chunk_create_table(chunk, ht, tablespace);
+	/* Insert chunk */
+	ts_chunk_insert_lock(chunk, RowExclusiveLock);
 
-	if (!OidIsValid(chunk->table_id))
-		elog(ERROR, "could not create chunk table");
+	/* Add metadata for dimensional and inheritable constraints */
+	ts_chunk_constraints_insert_metadata(chunk->constraints);
+}
 
+static void
+chunk_create_table_constraints(Chunk *chunk)
```

Comment:
Can the object be `const` here (or won't `ts_trigger_create_all_on_chunk` allow it)? I guess at least the pointer could be `const`, if that's important.

```suggestion
chunk_create_table_constraints(const Chunk *chunk)
```

Reply:
It doesn't allow to use `const` here due to string conversion by a PG function deeply insde..